cmd/object-handlers.go

```diff
@@ -779,7 +894,42 @@ func (api objectAPIHandlers) PutObjectHandler(w http.ResponseWriter, r *http.Req
 		}
 	}
 
-	hashReader, err := hash.NewReader(reader, size, md5hex, sha256hex)
+	var hashError error
+	actualSize := size
+	// Disabling compression for encrypted enabled requests.
+	// Using compression and encryption together enables room for side channel attacks.
+	// Eliminate non-compressible objects by extensions/content-types.
+	if !hasSSECustomerHeader(r.Header) && objectAPI.IsCompressionSupported() && !excludeForCompression(bucket, object, r.Header) && size > 0 {
```

Comment:
A complex `if`  can this be converted to a function?

Reply:
added a function `isCompressible(header, bucket, object) bool` which returns true for `compressible objects`

Reply:
Done